scripts/hid_configurator/configurator.py

```diff
@@ -231,30 +280,37 @@ def exchange_feature_report(dev, recipient, event_id, event_data, is_fetch):
 
 
 def get_device_pid(device_type):
-    if device_type in DEVICE_PID:
-        return DEVICE_PID[device_type]
-    else:
-        return None
+    return DEVICE[device_type]['pid']
+
+
+def get_device_vid(device_type):
+    return DEVICE[device_type]['vid']
 
 
 def open_device(device_type):
-    if get_device_pid(device_type) is None:
-        print("Unsupported device type")
-        return None
+    dev = None
 
-    try:
-        dev = hid.Device(vid=NORDIC_VID, pid=get_device_pid(device_type))
-        print("Found device")
-    except hid.HIDException:
+    devlist = [device_type]
+    if 'dongle' not in devlist:
+        devlist.append('dongle')
+
+    for i in devlist:
         try:
-            dev = hid.Device(vid=NORDIC_VID, pid=get_device_pid('dongle'))
-            print("Try to connect via dongle")
+            vid=get_device_vid(i)
+            pid=get_device_pid(i)
+            dev = hid.Device(vid=vid, pid=pid)
+            break
         except hid.HIDException:
-            print("Cannot find selected device nor dongle")
-            return None
-    except:
-        logging.error("Unknown error: {}".format(sys.exc_info()[0]))
-        return None
+            pass
+        except:
+            logging.error('Unknown error: {}'.format(sys.exc_info()[0]))
```

Comment:
Consider using `except Exception as e:` and then just print or log it ?

Reply:
Prefer to leave as is. Catching to a specific object would have sense if I would use it later. At the moment code (which was already there by the way) is showing exec stack and does not do anything with exception.